apps/orders/views.py

```diff
@@ -1,3 +1,102 @@
-from django.shortcuts import render
+from drf_util.utils import gt
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.serializers import Serializer
+from rest_framework.viewsets import GenericViewSet, mixins
+from rest_framework import status
 
-# Create your views here.
+from apps.common.permisions import IsAdmin
+from apps.orders.models import Order, Cart
+from apps.orders.serializers import OrderSerializer, CartSerializer, CartItemSerializer, CartDetailsSerializer, \
+    OrderStatusSerializer
+from apps.users.models import User
+from config.settings import env
+
+
+class OrderViewSet(GenericViewSet, mixins.ListModelMixin, mixins.CreateModelMixin, mixins.DestroyModelMixin):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    filterset_fields = ('user',)
+
+    def get_queryset(self):
+        qs = self.queryset
+
+        if hasattr(self, 'swagger_fake_view'):
+            qs = self.queryset.none()
+
+        if gt(self.request.user, 'role') == User.Role.USER:
+            qs = self.queryset.filter(user=self.request.user)
+
+        return qs
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+    @action(detail=True, methods=['POST'], serializer_class=Serializer)
+    def cancel(self, request, *args, **kwargs):
+        instance = self.get_object()
+        instance.status = Order.Status.CANCELED
+        instance.save()
+        return Response(status.HTTP_200_OK)
+
+    @action(detail=True, methods=['PATCH'], serializer_class=OrderStatusSerializer,
```

Comment:
nu este nevoie de action aparte, poti folosi mixinul de patch care este default in django